.circleci/config.yml

```diff
@@ -0,0 +1,452 @@
+      #     working_directory: girder_build
+      #    environment:
+      #      JASMINE_TIMEOUT: 15000
+      - run:
+          name: Rename javascript coverage files to avoid conflicts
+          command: for file in *.cvg; do mv "$file" "${file%.cvg}integration.cvg"; done
+          working_directory: girder/build/test/coverage/web_temp
+      - persist_to_workspace:
+          root: /home/circleci/project
+          paths:
+            - girder/.coverage.*
+            - girder/build/test/coverage/web_temp/*.cvg
+
+  py2_coverage:
+    docker:
+      # This image uses the oldest version of many dependencies
+      - image: girder/girder_test_py2:latest
+      # Use the oldest supported MongoDB
+      - image: mongo:2.6
+    working_directory: /home/circleci/project # as $CIRCLE_WORKING_DIRECTORY
+    steps:
+      - attach_workspace:
+          at: /home/circleci/project
+      - run:
+          name: Activate virtual environment
+          command: echo "source $CIRCLE_WORKING_DIRECTORY/girder_env/bin/activate" >> $BASH_ENV
+      # - run:
+      #     # We should be able to alter the ci_test.cmake to do this
+      #     name: Run CMake
+      #     command: cmake ../girder -DPYTHON_VERSION=3.4 -DPYTHON_EXECUTABLE=/home/circleci/project/girder_env/bin/python -DVIRTUALENV_EXECUTABLE=/home/circleci/project/girder_env/bin/virtualenv
+      - run:
+          name: Run CTest
+          command: ctest -VV -S "$CIRCLE_WORKING_DIRECTORY/girder/.circleci/ci_test.cmake"
+          environment:
+            TEST_GROUP: coverage
+            PYTHON_VERSION: 2.7
+            PYTHON_EXECUTABLE: /home/circleci/project/girder_env/bin/python
+            JASMINE_TIMEOUT: 15000
+            BUILD_JAVASCRIPT_TESTS: OFF
+          working_directory: girder_build
+      - run:
+          name: Install Codecov client
+          command: pip install codecov
+      - run:
+          name: Upload coverage
+          command: codecov --disable search pycov gcov --file $CIRCLE_WORKING_DIRECTORY/girder/build/test/coverage/server.xml
+          working_directory: girder
+
+  py3_integrationTests:
+    machine: true
+    working_directory: /home/circleci/project # as $CIRCLE_WORKING_DIRECTORY
+
+    steps:
+      - checkout:
+          path: girder
+      - run:
+          name: Get rid of existing versions of Mongo
+          command: sudo rm -f /etc/apt/sources.list.d/*mongo* && sudo apt-get remove mongodb-org mongodb-org-tools mongodb-org-shell mongodb-org-server mongodb-org-mongos || true
+      - run:
+          # This has to be done in a python 2 context
+          name: install node-gyp
+          command: npm install -g node-gyp
+      - run:
+          name: allow boto to work with python 3
+          command: sudo rm -f /etc/boto.cfg
+      - run:
+          name: Set up Python 3.5
+          command: |
+             # 3.5.2 is already installed in circle, so use it
+             pyenv install 3.5.2 || true
```

Comment:
You can specify any python version you like, but if it is already installed it throws an error.  We should try to install a particular version of python, since if Circle changes their python version, we would otherwise fail, but as long as it exists, this is effectively a noop that returns false.  Using the version they have speeds up the process.